share/functions/__fish_print_hostnames.fish

```diff
@@ -1,16 +1,23 @@
 
 function __fish_print_hostnames -d "Print a list of known hostnames"
-    # HACK: This only deals with ipv4
-
     # Print all hosts from /etc/hosts
     # use 'getent hosts' on OSes that support it (OpenBSD and Cygwin do not)
-    if type -q getent
-        and getent hosts >/dev/null 2>&1 # test if 'getent hosts' works and redirect output so errors don't print
+    if type -q getent; and getent hosts > /dev/null 2>&1 # test if 'getent hosts' works and redirect output so errors don't print
         # Ignore zero ips
-        getent hosts | string match -r -v '^0.0.0.0' | string replace -r '[0-9.]*\s*' '' | string split " "
+        getent hosts | string match --regex --invert '^0.0.0.0' \
+            # Remove left addresses column
+            | string replace --regex '^\s*\S+\s+' '' \
+            # Tokenize remaining hostnames and aliases columnss
+            | string split ' '
     else if test -r /etc/hosts
-        # Ignore commented lines and functionally empty lines. Strip comments.
-        string match -r -v '^\s*0.0.0.0|^\s*#|^\s*$' </etc/hosts | string replace -ra '#.*$' '' | string replace -r '[0-9.]*\s*' '' | string trim | string replace -ra '\s+' '\n'
+        # Ignore commented lines and functionally empty lines
```

Comment:
Please, do not mix comments and statements in this way. Not just here but everywhere else you've done this. It is confusing, breaks `fish_indent`, and the behavior is likely to change in the future. Also, such a complicated sequence of operations should be done in a discrete function that is invoked here.